cetmix_tower_server/models/cx_tower_reference_mixin.py

```diff
@@ -68,13 +69,11 @@ def _generate_or_fix_reference(self, reference_source):
         final_reference = reference
 
         # If exclude same records from search results
-        if self:
+        if self and not self.env.context.get("reference_mixin_skip_self"):
```

Comment:
Why do we need to skip it?

Reply:
when we copy the value of a variable, we cannot verify the uniqueness, because this point is added to the domain of another condition

Reply:
Sorry, don't understand this quite well. Could you please elaborate it further with some examples?

Reply:
when we copy
```python
variable_value.copy({"plan_line_action_id": new_action.id})
```
we can't check its reference for uniqueness
because here (line 72) in self (old_record)
we got domain where domain = [("id", "not in", self.ids), ("reference","=", reference)] and it sometimes incorrect

Reply:
What does it mean "incorrect"?

Reply:
Maybe I expressed myself incorrectly. What I meant was that if the search domain is like this, we will not be able to get a reliable result because the domain limits the choice of data.

Reply:
Ok, understood